sources/packages/api/src/services/student-scholastic-standings/student-scholastic-standings.service.ts

```diff
@@ -102,73 +135,277 @@ export class StudentScholasticStandingsService extends RecordDataModelService<St
+   * restriction add SSR restriction.
+   * When institution reports a change related to a FT application for unsuccessful
+   * completion and the total number of unsuccessful weeks hits minimum 68, add SSR
+   * restriction.
+   * If a ministry user resolves the SSR or WTHD restriction, and new withdrawal
+   * is reported, re add the above restrictions.
+   * If a ministry user resolves the SSR restriction, and new unsuccessful completion
+   * is reported, add the restriction (minimum is still at least 68).
+   * @param scholasticStandingData scholastic standing data.
+   * @param studentId student id.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   * @param applicationId application id.
+   * @returns a new student restriction object, that need to be saved.
+   */
+  async getFullTimeStudentRestrictions(
+    scholasticStandingData: ScholasticStanding,
+    studentId: number,
+    auditUserId: number,
+    applicationId: number,
+  ): Promise<StudentRestriction | undefined> {
+    if (
+      scholasticStandingData.scholasticStanding ===
+      SCHOLASTIC_STANDING_STUDENT_DID_NOT_COMPLETE_PROGRAM
+    ) {
+      if (!scholasticStandingData.numberOfUnsuccessfulWeeks) {
+        throw new Error(`Number of unsuccessful weeks is empty.`);
+      }
+      const totalExistingUnsuccessfulWeeks =
+        await this.getTotalFullTimeUnsuccessfulWeeks(studentId);
+
+      // When total number of unsuccessful weeks hits minimum 68, add SSR restriction.
+      if (
+        totalExistingUnsuccessfulWeeks +
+          scholasticStandingData.numberOfUnsuccessfulWeeks >=
+        MINIMUM_UNSUCCESSFUL_WEEKS
+      ) {
+        return this.studentRestrictionService.createRestrictionToSave(
+          studentId,
+          RestrictionCode.SSR,
+          auditUserId,
+          applicationId,
+        );
+      }
+    }
+    if (
+      scholasticStandingData.scholasticStanding ===
+      SCHOLASTIC_STANDING_STUDENT_WITHDREW_FROM_PROGRAM
+    ) {
+      // Check if "WTHD" restriction is already present for the student,
+      // if not add "WTHD" restriction else add "SSR" restriction.
+      const isWTHDAlreadyExists =
+        await this.studentRestrictionService.studentHasRestriction(
+          studentId,
+          RestrictionCode.WTHD,
+        );
+
+      const restrictionCode = isWTHDAlreadyExists
+        ? RestrictionCode.SSR
+        : RestrictionCode.WTHD;
+
+      return this.studentRestrictionService.createRestrictionToSave(
+        studentId,
+        restrictionCode,
+        auditUserId,
+        applicationId,
+      );
+    }
+  }
+
+  /**
+   * Get part time related restrictions for scholastic standing.
+   * When institution report Withdrawal OR unsuccessful for a PT course application,
+   * add PTSSR restriction to student.
+   * If a ministry user resolves the PTSSR restriction, and new withdrawal/unsuccessful
+   * is reported, re add the above restrictions.
```

Comment:
I did not really get what you were trying to explain here.

Reply:
Please ignore my comment here.

Reply:
for a part-time application, if there is a withdrawal or unsuccessful reported, without checking anything we will directly add `PTSSR` restriction. if the ministry user resolves this `PTSSR ` for the student, and a new withdrawal or unsuccessful reported for the student, we will again add a new `PTSSR` restriction.